src/invoke.rs

```diff
@@ -109,10 +101,14 @@ pub mod snapshot {
             }
         }
 
-        for (lk, le) in new_state {
-            lk.write_xdr(&mut file)?;
-            le.write_xdr(&mut file)?;
-        }
+        let vec_new_state: VecM<(LedgerKey, LedgerEntry)> = new_state
+            .into_iter()
+            .map(|x| (x.0, x.1))
+            .collect::<Vec<(LedgerKey, LedgerEntry)>>()
+            .try_into()?;
```

Comment:
The map isn't required because an OrdMap when iterated is already a tuple of key and value. 
```suggestion
        let vec_new_state: VecM<_> = new_state
            .into_iter()
            .collect::<Vec<_>>()
            .try_into()?;
```

Reply:
Good point. Done.